libs/dal/src/repositories/topic/topic.repository.ts

```diff
@@ -0,0 +1,30 @@
+import { AuthProviderEnum } from '@novu/shared';
+import { FilterQuery } from 'mongoose';
+
+import { EnvironmentId, OrganizationId, TopicKey, TopicEntity } from './topic.entity';
+import { Topic } from './topic.schema';
+
+import { BaseRepository } from '../base-repository';
+
+type IPartialTopicEntity = Omit<TopicEntity, '_environmentId' | '_organizationId'>;
+
+type EnforceEnvironmentQuery = FilterQuery<IPartialTopicEntity> &
+  ({ _environmentId: EnvironmentId } | { _organizationId: OrganizationId });
+
+export class TopicRepository extends BaseRepository<EnforceEnvironmentQuery, TopicEntity> {
+  constructor() {
+    super(Topic, TopicEntity);
+  }
+
+  async findTopicByKeyOrganizationAndEnvironment(
```

Comment:
```suggestion
  async findTopicByKey(
```

Since we are using environment and organization in every request because we are a multi tenant. I would suggest to omit this from the name, otherwise we need to add it to all of our methods and it will be quite tedious :)

Also I think it will be shorter and more understandable. Wdyt?